### PR TITLE
MSDKUI-1533 Test display current speed and speed limit warning

### PR DESCRIPTION
**Pull request by Docik**
Test implementation and simulation speed management

Signed-off-by: Aleksandr Dotsenko <1225028+Docik@users.noreply.github.com>

### CODE REVIEW

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreMatchers.kt [hunk 1]
```diff
@@ -145,6 +150,50 @@ object CoreMatchers {
...
+                }
+                return true
+            }
+        }
+    }
+
+    /**
+     * Matches [TextView] based on it's attribute color.
+     */
+    @Beta
```
Comment (tstaszak89) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreMatchers.kt]: Why is this @Beta annotation here?

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreMatchers.kt [hunk 2]
```diff
@@ -145,6 +150,49 @@ object CoreMatchers {
...
+    fun isNumeric():Matcher<View> {
+        return object: TypeSafeMatcher<View>() {
+            override fun describeTo(description: Description) {
+                description.appendText("is numeric ")
+            }
+
+            override fun matchesSafely(item: View): Boolean {
+                with(item as TextView) {
+                    try {
+                        parseDouble(text.toString())
```
Comment (imvjk) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreMatchers.kt]: this seems kind of hack.. I would go with some regex.

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreMatchers.kt [hunk 3]
```diff
@@ -145,6 +150,49 @@ object CoreMatchers {
...
+                        return false
+                    }
+                }
+                return true
+            }
+        }
+    }
+
+    /**
+     * Matches [TextView] based on it's attribute color.
```
Comment (imvjk) [inline MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreMatchers.kt]: In general please provide the proper description for methods including params

File: MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/routepreview/RoutePreviewFragmentPresenter.kt [hunk 4]
```diff
@@ -39,6 +40,7 @@ class RoutePreviewFragmentPresenter() : BasePresenter<GuidanceContracts.RoutePre
 
     private var state = State()
     var provider = Provider()
+    var simulationSpeed: Long = 0
```
Comment (imvjk) [inline MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/routepreview/RoutePreviewFragmentPresenter.kt]: so why do u need this?

Comment (imvjk) [inline MSDKUIDemo/MSDKUIApp/src/main/java/com/here/msdkuiapp/common/routepreview/RoutePreviewFragmentPresenter.kt]: whenever u need some changes in real code please create a ticket for that. that should go with a separate review

File: MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreMatchers.kt (hunk 3 above)
Comment (Docik) [reply MSDKUIDemo/MSDKUIApp/src/androidTest/java/com/here/msdkuiapp/espresso/impl/core/CoreMatchers.kt]: Fixed

**Pull request merged**